app/src/main/java/com/duckduckgo/app/tabs/model/TabDataRepository.kt

```diff
@@ -51,6 +53,25 @@ class TabDataRepository @Inject constructor(
         return tabId
     }
 
+    override suspend fun addWithSource(url: String?, skipHome: Boolean, isDefaultTab: Boolean): String {
+        val tabId = generateTabId()
+        var sourceTabId: String? = null
+
+        withContext(Dispatchers.IO) {
```

Comment:
👍 you can perfectly use `withContext` here to change threads.

Because you mentioned that you were a bit new to coroutines, just a small nice tip 🙌.

If you check the signature of `withContext`:
```
public suspend fun <T> withContext(
    context: CoroutineContext,
    block: suspend CoroutineScope.() -> T
): T
```
That means `withContext` can return the value you are querying in the internal `block`.

You can do this:
```
        val sourceTabId = withContext(Dispatchers.IO) {
           tabsDao.selectedTab()?.tabId
        }
```

Reply:
Awesome, this is clean too!